samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -26,12 +26,10 @@
 
 
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
-
-
   synchronized public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
     ZkConfig zkConfig = new ZkConfig(config);
     ZkClient zkClient = new ZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
-    ZkUtils zkUtils = new ZkUtils(participantId, new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs());
```

Comment:
> a) I would prefer to use processorId in the constructor parameter here, with more precise meaning of what's the participant in this particular context.

I would defer to @sborya  on whether to call it processorId or participantId. You can add this question here - [SAMZA-1173](https://issues.apache.org/jira/browse/SAMZA-1173)

> as long as the ZNode path is not affected by removing the input parameter here, it should be fine.

It shouldn't be affected.

> Non-path related parameters usually occurs at the end.
That's a good suggestion. Seems like a good pattern to follow :)